File: app/src/main/java/de/tum/in/tumcampusapp/models/tumo/TelSubstationList.java
package de.tum.in.tumcampusapp.models.tumo;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.io.Serializable;
import java.util.List;

/**
 * Wrapper class holding a list of {@link TelSubstation}s. Note: This model is
 * based on the TUMOnline web service response format for a corresponding
 * request.
 */
@Root(name = "telefon_nebenstellen")
public class TelSubstationList implements Serializable {

    private static final long serialVersionUID = -3790189526859194869L;

    @ElementList(inline = true, required = false)
    private List<TelSubstation> substations;

    public List<TelSubstation> getSubstations() {
        return substations;
    }

    public void setSubstations(List<TelSubstation> substations) {
        this.substations = substations;
    }

}

File: app/src/main/java/de/tum/in/tumcampusapp/models/tumo/TelSubstationList.kt
package de.tum.`in`.tumcampusapp.models.tumo

import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

import java.io.Serializable

/**
 * Wrapper class holding a list of [TelSubstation]s. Note: This model is
 * based on the TUMOnline web service response format for a corresponding
 * request.
 */
@Root(name = "telefon_nebenstellen")
data class TelSubstationList(@field:ElementList(inline = true, required = false)
                             var substations: List<TelSubstation> = mutableListOf()) : Serializable {
    companion object {
        private const val serialVersionUID = -3790189526859194869L
    }
}
